pandas/io/excel/_base.py

```diff
@@ -961,6 +961,10 @@ class ExcelWriter(metaclass=abc.ABCMeta):
     # You also need to register the class with ``register_writer()``.
     # Technically, ExcelWriter implementations don't need to subclass
     # ExcelWriter.
+
+    _engine: str
+    _supported_extensions: tuple[str, ...] | list[str]
```

Comment:
Any reason not to type-hint this as just a tuple? I think we really don't want to make this a list; that would be mutable allowing the user to change via public methods.

Reply:
I just re-used the existing annotations from `supported_extensions`. Will change it to `tuple`

Reply:
Changed it!

It would be great to make `engine` and `supported_extensions` a classmethod+property to be able to access it even without having to instantiate it. But that seems complicated with [pre python 3.9](https://stackoverflow.com/a/1800999)

Reply:
Agreed - but given the difficulty I think it might be best to wait until we drop support for 3.8. I think this should be happening shortly after the release of Python 3.11.